src/controller/BUILD.gn

```diff
@@ -24,41 +24,91 @@ source_set("nodatamodel") {
   public_deps = [ "${chip_root}/src/app" ]
 }
 
+CHIP_CONTROLLER_SOURCES = [
+  "AbstractDnssdDiscoveryController.cpp",
+  "AutoCommissioner.cpp",
+  "CHIPCommissionableNodeController.cpp",
+  "CHIPDeviceControllerFactory.cpp",
+  "CHIPDeviceControllerFactory.h",
+  "CommissioneeDeviceProxy.cpp",
+  "CommissionerDiscoveryController.cpp",
+  "CommissionerDiscoveryController.h",
+  "CommissioningDelegate.cpp",
+  "ExampleOperationalCredentialsIssuer.cpp",
+  "ExampleOperationalCredentialsIssuer.h",
+  "SetUpCodePairer.cpp",
+]
+
+CHIP_READ_CLIENT_SOURCES = [
```

Comment:
silly q - what is a read client?

Reply:
I went off the "chip_enable_read_client" config variable. However this is odd to me ... it seems like we would have ReadClient.h/cpp from `src/app` which probably needs some things from here.

However overall I think this is seriously misguided: if I need `ReadClient` I should somewhere in the source tree have a `foo/bar:read_client` dependency rather than this magic. I believe it works around things in src/app not being covered by gn layering and dependencies at all. This is the bit that I try to unravel (and it is painful)